How to add dependencies in more detail (#1021)
pypa/hatch: docs/environment.md

Hatch ensures that environments are always compatible with the currently defined [project dependencies](config/metadata.md#dependencies) (if [installed](config/environment/overview.md#skip-install) and in [dev mode](config/environment/overview.md#dev-mode)) and [environment dependencies](config/environment/overview.md#dependencies).

> To add `cowsay` as a dependency, open `pyproject.toml` and add it to the `dependencies` list there:

Reviewer: ```suggestion
To add `cowsay` as a dependency, open `pyproject.toml` and add it to the [`dependencies`](config/metadata.md#dependencies) array:
```

> ```toml

Reviewer: ```suggestion
```toml tab="pyproject.toml"
[project]
...
```

dependencies = [
  "cowsay"
]
```

> This dependency will be installed the next time you use the `hatch run` command. For example:

Reviewer: ```suggestion

This dependency will be installed the next time you [spawn a shell](#entering-environments) or [run a command](#command-execution). For example:
```

                    ||----w |
                    ||     ||
```

> Note the "Syncing dependencies" line there, output by Hatch when it installed the new dependency that was missing from the environment.

Reviewer: ```suggestion

!!! note
    The `Syncing dependencies` status will display temporarily when Hatch updates environments in response to any dependency changes that you make.
```
